avahi/avahi.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package avahi
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/presotto/go-mdns-sd"
+)
+
+var logger *log.Logger
+
+type mdnsScanner interface {
+	ScanInterfaces() (string, error)
+}
+
+var _mdns mdnsScanner
+
+var initOnce sync.Once
+
+const hostnameDefault = "snapweb"
```

Comment:
haha, indeed. The problems of copy and paste. I've set "device" as the default one.